core/src/main/java/org/mskcc/cbio/portal/model/CanonicalGene.java

```diff
@@ -56,19 +57,39 @@
     private String type;
 
     public CanonicalGene(String hugoGeneSymbol) {
-        this(-1, hugoGeneSymbol);
```

Comment:
this is the part of the code that I don't like so much...I spent quite some time thinking of how to fix this part of the code...but perhaps we need to do this in a subsequent PR. For now, I focused on getting it to work with the new data model. One of the "problems" we have now is that genetic entity id is not always know, needed or used, depending on the data type or the process that is using it. E.g. 
- when importing new genes batch wise (not so often) we don't have a genetic entity yet
- when making a new "gene" on the fly for some entities like "phosphoproteins" (somewhat more often) we don't have a genetic entity yet
- when retrieving mutation data we don't need genetic entity id since all mutation data is still linked directly to entrez gene id. This also applies to discrete CNA data (via `cna_event` table)
- when retrieving genetic alteration data (such as RPPA, mRNA, etc) then genetic entity is needed

I'm therefore, a.o., wondering whether genetic entity id should be part of `CanonicalGene`. Maybe we should later on split this up in different classes to make such distinctions more explicit...